libraries/AC_WPNav/AC_Loiter.h

```diff
@@ -15,65 +15,115 @@ class AC_Loiter
+    // Result is returned in centimeters.
+    // See get_stopping_point_NE_m() for full details.
+    void get_stopping_point_NE_cm(Vector2f& stopping_point_ne_cm) const;
 
-    /// get vector to stopping point based on a horizontal position and velocity
-    void get_stopping_point_NE_cm(Vector2f& stopping_point) const;
+    // Calculates the expected stopping point based on current velocity and position in the NE frame.
+    // Result is returned in meters.
+    // Uses the position controller’s deceleration model.
+    void get_stopping_point_NE_m(Vector2f& stopping_point_ne_m) const;
 
-    /// get horizontal distance to loiter target in cm
-    float get_distance_to_target_cm() const { return _pos_control.get_pos_error_NE_cm(); }
+    // Returns the horizontal distance to the loiter target in centimeters.
+    // See get_distance_to_target_m() for full details.
+    float get_distance_to_target_cm() const { return get_distance_to_target_m() * 100.0; }
 
-    /// get bearing to target in centi-degrees
+    // Returns the horizontal distance to the loiter target in meters.
+    // Computed using the NE position error from the position controller.
+    float get_distance_to_target_m() const { return _pos_control.get_pos_error_NE_m(); }
+
+    // Returns the bearing from current position to the loiter target in radians.
+    // Bearing is relative to true north, using the NE frame.
     float get_bearing_to_target_rad() const { return _pos_control.get_bearing_to_target_rad(); }
 
-    /// get maximum lean angle when using loiter
-    float get_angle_max_rad() const;
+    // Returns the maximum pilot-commanded lean angle in centidegrees.
+    // See get_angle_max_rad() for full details.
     float get_angle_max_cd() const;
 
-    /// run the loiter controller
+    // Returns the maximum pilot-commanded lean angle in radians.
+    // - If `_angle_max_deg` is zero, this returns 2/3 of the limiting PSC angle.
+    // - Otherwise, returns the minimum of `_angle_max_deg` and PSC’s configured angle limit.
+    float get_angle_max_rad() const;
+
+    // Runs the loiter control loop, computing desired acceleration and updating position control.
+    // If `avoidance_on` is true, velocity is adjusted using avoidance logic before being applied.
     void update(bool avoidance_on = true);
 
-    //set maximum horizontal speed
+    // Sets the maximum allowed horizontal loiter speed in cm/s.
+    // See set_speed_max_NE_ms() for full details.
     void set_speed_max_NE_cms(float speed_max_NE_cms);
 
-    /// get desired roll, pitch which should be fed into stabilize controllers
-    float get_roll_rad() const { return _pos_control.get_roll_rad(); }
-    float get_pitch_rad() const { return _pos_control.get_pitch_rad(); }
+    // Sets the maximum allowed horizontal loiter speed in m/s.
+    // Internally converts to cm/s and clamps to a minimum of LOITER_SPEED_MIN_CMS.
+    void set_speed_max_NE_ms(float speed_max_NE_ms);
```

Comment:
nitpick: there's a slight difference in capitalisation of the argument between the .h and the .cpp.  It's "speed_max_NE_ms" in the .h and "speed_max_ne_ms" in the cpp.

Let's fix this in a follow-up PR (if we fix it at all)